examples/idempotency/src/bring_your_own_persistent_store.py

```diff
@@ -27,9 +27,7 @@ def __init__(
         boto_config: Optional[Config] = None,
         boto3_session: Optional[boto3.session.Session] = None,
     ):
-        boto_config = boto_config or Config()
-        session = boto3_session or boto3.session.Session()
-        self._ddb_resource = session.resource("dynamodb", config=boto_config)
+        self._ddb_resource = (boto3_session or boto3.session.Session()).resource("dynamodb", config=boto_config)
```

Comment:
I think that splitting this into two lines makes the code more "understandable" for someone reading it for the first time. What do you think?

```suggestion
        self._boto3_session = boto3_session or boto3.session.Session()
        self._ddb_resource = resource("dynamodb", config=self._boto3_session)
```

Reply:
Good point. Done! I'm not assigning them to `self` as instance attributes because we don't need the references to either boto3_session or boto3_client later.

As a side note, there might other unneeded assignments like these in the code, which communicate the wrong message of "we need this reference later" when we don't. In the worst case, this potentially contributes to memory leaks. Also, the use of "non-private" names like `self.client` instead of `self._client` in other parts of the code communicate that these can be publicly used, when in some cases I'm not sure they should be. But this is all a separate discussion with too many breaking changes involved, so probably too late now to go over all this...